core/src/main/scala/kafka/common/ZkNodeChangeNotificationListener.scala

```diff
@@ -58,6 +59,7 @@ class ZkNodeChangeNotificationListener(private val zkUtils: ZkUtils,
   def init() {
     zkUtils.makeSurePersistentPathExists(seqNodeRoot)
     zkUtils.zkClient.subscribeChildChanges(seqNodeRoot, NodeChangeListener)
+    zkUtils.zkClient.subscribeStateChanges(ZkStateChangeListener)
     processAllNotifications()
```

Comment:
By the time this class's init method is called the session may or may not have initialized. if its already initialized this class's  ZkStateChangeListener'handleNewSession method will only be called if a reconnection happens or if a child changes. To avoid that delay we call the processAllNotifications() as part of init.

If the zkClient is not initialized, I believe all the calls block at ZkClient level for a session to be established so it should not be an issue.